test/Test/WebDriver/WebDriverTest.php

```diff
@@ -221,4 +225,22 @@ public function testSeleniumNoResponse()
 
         $this->assertEquals(null, $out);
     }
+
+    /**
+     * Assert that empty response does not trigger exception, but invalid JSON does
+     */
+    public function testNonJsonResponse()
+    {
+        ServiceFactory::getInstance()->setServiceClass('service.curl', '\\Test\\WebDriver\\TestCurlService');
```

Comment:
Yes also, but I was hesitant to make any strong changes to an unrelated API. ServiceFactory would be good if you could inject a mock directly.